src/main/java/com/alipay/oceanbase/hbase/OHTable.java

```diff
@@ -740,18 +778,42 @@ public boolean checkAndPut(byte[] row, byte[] family, byte[] qualifier,
 
     private void innerDelete(Delete delete) throws IOException {
         checkArgument(delete.getRow() != null, "row is null");
-        checkArgument(!delete.isEmpty(), "delete is empty");
         List<Integer> errorCodeList = new ArrayList<Integer>();
+        BatchOperationResult results = null;
+        
         try {
             checkFamilyViolation(delete.getFamilyMap().keySet());
+            if (delete.getFamilyMap().isEmpty()) {
+                KeyValue kv = new KeyValue(delete.getRow(), delete.getTimeStamp(),
+                        KeyValue.Type.DeleteFamily);
+                
+                BatchOperation batch = buildBatchOperation(tableNameString, Arrays.asList(kv), false, null);
+                results = batch.execute();
+            } else if (delete.getFamilyMap().size() > 1) {
+                boolean has_delete_family = delete.getFamilyMap().entrySet().stream()
+                        .flatMap(entry -> entry.getValue().stream())
+                        .anyMatch(kv -> KeyValue.Type.codeToType(kv.getType()) == KeyValue.Type.DeleteFamily);
+                if (!has_delete_family) {
+                    BatchOperation batch = buildBatchOperation(tableNameString,
+                            delete.getFamilyMap(), false, null);
+                    results = batch.execute();
+                } else {
+                    for (Map.Entry<byte[], List<KeyValue>> entry : delete.getFamilyMap().entrySet()) {
```

Comment:
Why can't we handle multi-column family deletion which include DeleteFamily type currently?

Reply:
added comments to explain this issue